skbio/stats/tests/test_composition.py

```diff
@@ -269,6 +335,85 @@ def test_centralize(self):
+        result = ancom(pd.DataFrame(self.table1),
+                       pd.Series(self.cats1),
+                       multiple_comparisons_correction=None)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([6, 6, 2, 2, 2, 2, 2]))
+        npt.assert_allclose(reject.values,
+                            np.array([True, True, False, False,
+                                      False, False, False], dtype=bool))
+        result = ancom(self.table1, self.cats1,
+                       multiple_comparisons_correction=None)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([6, 6, 2, 2, 2, 2, 2]))
+        npt.assert_allclose(reject.values,
+                            np.array([True, True, False, False,
+                                      False, False, False], dtype=bool))
+
+        result = ancom(self.table1,
+                       self.cats1,
+                       multiple_comparisons_correction='holm-bonferroni',
+                       significance_test=scipy.stats.mannwhitneyu)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([0, 0, 0, 0, 0, 0, 0]))
+        npt.assert_allclose(reject.values,
+                            np.array([False]*7, dtype=bool))
+        result = ancom(self.table1,
+                       self.cats1,
+                       multiple_comparisons_correction=None,
+                       significance_test=scipy.stats.mannwhitneyu)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([6, 6, 2, 2, 2, 2, 2]))
+        npt.assert_allclose(reject.values,
+                            np.array([True,  True, False, False,
+                                      False, False, False], dtype=bool))
+
+        result = ancom(self.table2,
+                       self.cats2,
+                       multiple_comparisons_correction=None,
+                       significance_test=scipy.stats.mannwhitneyu)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([8, 8, 3, 3,
+                                      8, 3, 3, 3, 3]))
+        npt.assert_allclose(reject.values,
+                            np.array([True, True, False, False,
+                                      True, False, False, False, False],
+                                     dtype=bool))
+
+        result = ancom(self.table3,
+                       self.cats3,
+                       multiple_comparisons_correction=None)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([0]*7))
+        npt.assert_allclose(reject.values,
+                            np.array([False]*7))
+
+    def test_ancom_fail(self):
```

Comment:
Just added a test for this.

Now, it does give an error that may be hard to debug, namely
```
Traceback (most recent call last):
  File "test_composition.py", line 681, in test_ancom_fail_multiple_groups
    significance_test=scipy.stats.ttest_ind)
  File "/Users/mortonjt/Documents/software/python/scikit-bio/skbio/stats/composition.py", line 848, in ancom
    _logratio_mat = _log_compare(mat.values, cats.values, significance_test)
  File "/Users/mortonjt/Documents/software/python/scikit-bio/skbio/stats/composition.py", line 942, in _log_compare
    arr=ratio)
  File "/Users/mortonjt/miniconda/envs/skbio/lib/python3.4/site-packages/numpy/lib/shape_base.py", line 91, in apply_along_axis
    res = func1d(arr[tuple(i.tolist())], *args, **kwargs)
  File "/Users/mortonjt/Documents/software/python/scikit-bio/skbio/stats/composition.py", line 936, in func
    return significance_test(*[x[cats == k] for k in cs])
  File "/Users/mortonjt/miniconda/envs/skbio/lib/python3.4/site-packages/scipy/stats/stats.py", line 3546, in ttest_ind
    v1 = np.var(a, axis, ddof=1)
  File "/Users/mortonjt/miniconda/envs/skbio/lib/python3.4/site-packages/numpy/core/fromnumeric.py", line 3089, in var
    keepdims=keepdims)
  File "/Users/mortonjt/miniconda/envs/skbio/lib/python3.4/site-packages/numpy/core/_methods.py", line 79, in _var
    rcount = _count_reduce_items(arr, axis)
  File "/Users/mortonjt/miniconda/envs/skbio/lib/python3.4/site-packages/numpy/core/_methods.py", line 50, in _count_reduce_items
    items *= arr.shape[ax]
TypeError: only integer arrays with one element can be converted to an index
```

This error is being thrown when calling `scipy.stats.ttest_ind` on multiple groups.
Is this something that we should be handling (I can't think of a clean solution on the top of my head).